integr8ly/integreatly-operator
# Intly 2798: Dedicated admin to 3scale admin

Jira: https://issues.jboss.org/browse/INTLY-2798

3scale and rhsso part of the above ticket.
Openshift users should be automatically added to 3scale and rhsso and openshift users that are in the dedicated-admins group should be admins in 3scale and rhsso also.


## Verification
1. Add an OperatorSource that points to my app registry
```yaml
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: integreatly-operators
  namespace: openshift-marketplace
spec:
  authorizationToken: {}
  displayName: Integreatly Operators
  endpoint: 'https://quay.io/cnr'
  publisher: Integreatly Publisher
  registryNamespace: sedroche
  type: appregistry
```
2. Create an Installation CR and wait for the installation to succeed.
3. Go to the 3scale admin console (3scale namespace -> routes -> `https://3scale-admin.<routingSubdomain>`) and using the ` Authenticate through Red Hat Single Sign-On` option, sign in using an openshift user. (The sign-in should not need any extra configuration or ask for any extra verification)
4. If the openshift user is not part of the `dedicated-admins` group it should have limited options within the 3scale admin console
5. To test you can create the `dedicated-admins` group by:
```bash
oc adm groups new dedicated-admins
```
then add and remove users by:
```bash
oc adm groups add-users dedicated-admins <username>
```
and
```bash
oc adm groups remove-users dedicated-admins <username>
```
6. Adding and deleting openshift users should be reflected in the 3scale and rhsso users.

7. Adding and removing users from the `dedicated-admins` group should change roles in rhsso and 3scale
